src/main/java/eu/dissco/core/digitalspecimenprocessor/service/AnnotationPublisherService.java

```diff
@@ -54,39 +55,45 @@ private static OaHasSelector buildNewSpecimenSelector() {
   }
 
   public void publishAnnotationNewSpecimen(Set<DigitalSpecimenRecord> digitalSpecimenRecords) {
-    for (DigitalSpecimenRecord digitalSpecimenRecord : digitalSpecimenRecords) {
-      try {
-        var annotationProcessingRequest = mapNewSpecimenToAnnotation(digitalSpecimenRecord);
-        kafkaPublisherService.publishAcceptedAnnotation(
-            new AutoAcceptedAnnotation(
-                createMachineAgent(applicationProperties.getName(),
-                    applicationProperties.getPid(), PROCESSING_SERVICE, DOI,
-                    SCHEMA_SOFTWARE_APPLICATION),
-                annotationProcessingRequest));
-      } catch (JsonProcessingException e) {
-        log.error("Unable to send auto-accepted annotation for new specimen: {}",
-            digitalSpecimenRecord.id(), e);
-      }
+    try {
+      var annotationProcessingRequest = digitalSpecimenRecords.stream()
+          .map(this::mapNewSpecimenToAnnotation)
+          .filter(Objects::nonNull)
+          .toList();
+      kafkaPublisherService.publishAcceptedAnnotation(
+          new AutoAcceptedAnnotation(
+              createMachineAgent(applicationProperties.getName(),
+                  applicationProperties.getPid(), PROCESSING_SERVICE, DOI,
+                  SCHEMA_SOFTWARE_APPLICATION),
+              annotationProcessingRequest));
+    } catch (JsonProcessingException e) {
+      log.error("Unable to send auto-accepted annotation for new specimens", e);
     }
   }
 
   private AnnotationProcessingRequest mapNewSpecimenToAnnotation(
-      DigitalSpecimenRecord digitalSpecimenRecord) throws JsonProcessingException {
```

Comment:
I think before all these JsonProcessingException were thrown to the top level method and that would handle the ex. What was the reason behind changing this and handle it in all the individual methods?